tox_docker/tox4/log.py

```diff
@@ -0,0 +1,28 @@
+from logging import Logger
+from typing import Optional
+
+from tox.report import LOGGER, ToxHandler
+
+_HANDLER = None
+
+
+def _get_handler() -> ToxHandler:
+    global _HANDLER
+    if _HANDLER is None:
+        logger: Optional[Logger] = LOGGER
+        while logger:
+            for handler in logger.handlers:
+                if isinstance(handler, ToxHandler):
+                    _HANDLER = handler
+                    logger = None
+                    break
+            else:
+                logger = logger.parent if logger.propagate else None
+
+    assert _HANDLER is not None
+    return _HANDLER
+
+
+def log(line: str) -> None:
+    with _get_handler().with_context("docker"):
```

Comment:
We don't support this and will likely break horribly with journals. IMHO this should be attached to the env, as is by default.

Reply:
hm, I could swear when I first wrote this, just using a `logging.warning()` didn't print with the right format. it works now, so will clean this mess up.